charts/minibroker/templates/rbac.yaml

```diff
@@ -1,22 +1,91 @@
+---
 apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: minibroker
-  labels:
-    app: {{ template "fullname" . }}
-    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-    release: "{{ .Release.Name }}"
-    heritage: "{{ .Release.Service }}"
+  {{- template "minibroker.labels" . }}
+
+{{- if .Capabilities.APIVersions.Has "rbac.authorization.k8s.io/v1" }}
+
+# If a "defaultNamespace" has been defined, then only grant access to the
+# release namespace itself (for storing data in a configmap), and to the
+# defaultNamespace (to maintain the service instances).
+{{- if .Values.defaultNamespace }}
+
+# If we install the services into the release namespace, then the namespace
+# already exists, and the configmap role binding becomes redundant because
+# we'll grant full access to the defaultNamespace.
+{{- if ne .Release.Namespace .Values.defaultNamespace }}
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: {{ .Values.defaultNamespace }}
+  {{- template "minibroker.labels" . }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: minibroker-configmap
+  {{- template "minibroker.labels" . }}
+rules:
+- apiGroups: [""]
+  resources: ["configmaps"]
+  verbs:     ["*"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: minibroker-configmap
+  {{- template "minibroker.labels" . }}
```

Comment:
Does the role binding not need an NS here? (To match the cross-NS `minibroker-services` role binding)

Reply:
This rolebinding is inside the `.Release.Namespace`, which is the default for all objects that don't specify an explicit namespace. You could add

```
    namespace: {{ .Release.Namespace }}
```

but it would be redundant, so I prefer not to include it.